spec/features/settings/dex_connector_ldap_feature_spec.rb

```diff
@@ -134,67 +134,51 @@
   describe "#new_save_disabled_by_change" do
     before do
       visit new_settings_dex_connector_ldap_path
-      have_button("Save", disabled: true)
+      expect(page).to have_button("Save", disabled: true)
```

Comment:
I did test have_button by itself without the expect and it definitely works. Could you explain the difference with that versus adding the expect? Does the expect just repeat the test for the timeout period?

Reply:
> Could you explain the difference with that versus adding the expect?

The `have_button` matcher will only look for the element but won't make the test fail if doesn't find what it was looking for. The `expect` in conjunction with `to` or `not_to` will be the one matching your expectations and be responsible to throw an error if it's the case.

So, in this case, the test passed because we were not explicitly expecting `have_button` to be a truthy matcher. What will decide if it's a truthy or falsy matcher is the `to` and `not_to` from the `expect`.